lib/ri_cal/component/t_z_info_timezone.rb

```diff
@@ -21,20 +21,20 @@ def initialize(which, this_period, prev_period)
       @abbreviation = this_period.abbreviation
       @rdates = []
     end
-    
+
     def daylight?
       @which == "DAYLIGHT"
     end
-    
+
     def period_local_end(period)
-      (period.local_end || DateTime.parse("99990101T000000")).strftime("%Y%m%dT%H%M%S")
+      (period.local_ends_at&.to_datetime || DateTime.parse("99990101T000000")).strftime("%Y%m%dT%H%M%S")
```

Comment:
```suggestion
      (period.local_ends_at && period.local_ends_at.to_datetime || DateTime.parse("99990101T000000")).strftime("%Y%m%dT%H%M%S")
```

As `&.` is a Ruby 2.6 feature and this gem is really old and most likely used in old services.

If you insist on using it, you should add `spec.required_ruby_version = ">= 2.6.0"` to the _.gemspec_ and the version should be raised at least by a minor level.